conf/nginx/security.conf.inc

```diff
@@ -25,18 +25,29 @@ ssl_dhparam /usr/share/yunohost/ffdhe2048.pem;
 # Follows the Web Security Directives from the Mozilla Dev Lab and the Mozilla Obervatory + Partners
 # https://wiki.mozilla.org/Security/Guidelines/Web_Security
 # https://observatory.mozilla.org/
-{% if experimental == "True" %}
-more_set_headers "Content-Security-Policy : upgrade-insecure-requests; default-src https: data: blob: ; object-src https: data: 'unsafe-inline'; style-src https: data: 'unsafe-inline' ; script-src https: data: 'unsafe-inline' 'unsafe-eval'";
-{% else %}
-more_set_headers "Content-Security-Policy : upgrade-insecure-requests";
-more_set_headers "Content-Security-Policy-Report-Only : default-src https: data: blob: ; object-src https: data: 'unsafe-inline'; style-src https: data: 'unsafe-inline' ; script-src https: data: 'unsafe-inline' 'unsafe-eval'";
-{% endif %}
 more_set_headers "X-Content-Type-Options : nosniff";
 more_set_headers "X-XSS-Protection : 1; mode=block";
 more_set_headers "X-Download-Options : noopen";
 more_set_headers "X-Permitted-Cross-Domain-Policies : none";
 more_set_headers "X-Frame-Options : SAMEORIGIN";
 
+# Here we add a generic CSP headers only if the app doesn't define it.
+# NB: if the <meta tag is used the most restrictive rules will be used
+{% if experimental == "True" %}
+header_filter_by_lua_block {
+  if ngx.header["Content-Security-Policy"] == nil or ngx.header["Content-Security-Policy"] == "" then
+    ngx.header["Content-Security-Policy"] = "upgrade-insecure-requests; default-src https: data: blob: ; object-src https: data: 'unsafe-inline'; style-src https: data: 'unsafe-inline' ; script-src https: data: 'unsafe-inline' 'unsafe-eval'"
+  end
+}
+{% else %}
+header_filter_by_lua_block {
+  if ngx.header["Content-Security-Policy"] == nil or ngx.header["Content-Security-Policy"] == "" then
+    ngx.header["Content-Security-Policy"] = "upgrade-insecure-requests"
+  end
+}
+more_set_headers "Content-Security-Policy-Report-Only : default-src https: data: blob: ; object-src https: data: 'unsafe-inline'; style-src https: data: 'unsafe-inline' ; script-src https: data: 'unsafe-inline' 'unsafe-eval'";
```

Comment:
Content-Security-Policy-Report-Only is useless without a report-uri (and probably a report-to) directive, pointing to a host that accepts violation reports for your domain: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-uri

Reply:
Indeed feel free to remove it